java/code/src/com/suse/manager/xmlrpc/iss/HubHandler.java

```diff
@@ -388,15 +405,167 @@ public int deregister(User loggedInUser, String fqdn, boolean onlyLocal) {
+        try {
+            hubManager.updateServerData(loggedInUser, fqdn, IssRole.valueOf(role), data);
+        }
+        catch (IllegalArgumentException ex) {
+            throw new InvalidParameterException(logGetErrorString(ex, "Illegal state"));
+        }
+        return 1;
+    }
+
+    private void ensureValidRole(String role) throws PermissionCheckFailureException {
+        try {
+            IssRole.valueOf(role);
+        }
+        catch (IllegalArgumentException ex) {
+            throw new InvalidParameterException(logGetErrorString(
+                    role, "Invalid role, must either be HUB or PERIPHERAL"));
+        }
+    }
+
+    /**
+     * Collect data about a Manager Server
+     *
+     * @param loggedInUser the user
+     * @return a {@link ManagerInfoJson} on success, exception otherwise
+     * @apidoc.doc Get manager info.
+     * @apidoc.param #session_key()
+     * @apidoc.returntype $ManagerInfoJsonSerializer
+     */
+    @ReadOnly
+    public ManagerInfoJson getManagerInfo(User loggedInUser) {
+        ensureSatAdmin(loggedInUser);
+        try {
+            return hubManager.collectManagerInfo(loggedInUser);
+        }
+        catch (Exception ex) {
+            throw new InvalidParameterException(logGetErrorString(ex, "Error while collecting manager info"));
+        }
+    }
+
+    /**
+     * Collect data about all peripheral organizations
+     *
+     * @param loggedInUser the user
+     * @return a list of {@link Org} on success, exception otherwise
+     * @apidoc.doc Collect data about all peripheral organizations.
+     * @apidoc.param #session_key()
+     * @apidoc.returntype
+     *      #return_array_begin()
+     *          $OrgSerializer
+     *      #array_end()
+     */
+    @ReadOnly
+    public List<Org> listAllPeripheralOrgs(User loggedInUser) {
+        ensureSatAdmin(loggedInUser);
+        try {
+            return hubManager.collectAllOrgs(loggedInUser);
+        }
+        catch (Exception ex) {
+            throw new InvalidParameterException(logGetErrorString(ex,
+                    "Error while collecting all peripheral organization"));
+        }
+    }
+
+    /**
+     * Collect data about all peripheral channels
+     *
+     * @param loggedInUser the user
+     * @return a list of {@link Channel} on success, exception otherwise
+     * @apidoc.doc Collect data about all peripheral channels.
+     * @apidoc.param #session_key()
+     * @apidoc.returntype
+     *      #return_array_begin()
+     *          $ChannelSerializer
+     *      #array_end()
+     */
+    @ReadOnly
+    public List<Channel> listAllPeripheralChannels(User loggedInUser) {
+        ensureSatAdmin(loggedInUser);
+        try {
+            return hubManager.collectAllChannels(loggedInUser);
+        }
+        catch (Exception ex) {
+            throw new InvalidParameterException(logGetErrorString(ex,
+                    "Error while collecting all peripheral channels"));
+        }
```

Comment:
By the way, for one-liner `try-catch` like this one, there is an utility class, even though I think no one has ever used it since I introduced it. It's called `Exceptions`. You can do:

```java
        Exceptions.handleByWrapping(
            () -> hubManager.collectAllChannels(loggedInUser),
            (ex) -> new InvalidParameterException(logGetErrorString(ex, "Error while collecting all peripheral channels"))
        );
```

I'll leave it up to you to decide if it's better or worse to read :smile: